src/bom/bitcode.rs

```diff
@@ -263,9 +277,62 @@ fn build_bitcode_arguments(chan : &mut mpsc::Sender<Option<Event>>,
 {
     let mut orig_target = None;
 
+    // Determine the language of the original input file, so that the bitcode can
+    // utilize the same language, because using C++ mode to compile C will cause
+    // errors:
+    //
+    // char* greeting() { return "hello, world"; }
+    //
+    // The above is perfectly acceptable C, but if compile as C++ it yields:
+    // error: ISO C++11 does not allow conversion from string literal to 'char*'.
+    let is_c_plusplus = orig_args.iter().find(
+        |&arg| arg.to_str().map(|a| !a.starts_with("-") &&
+                                (a.ends_with(".cc") ||
+                                 a.ends_with(".cpp")))
+            .unwrap_or_else(|| false))
+        .is_some();
```

Comment:
Can we extract this into a separate function? There is a lot of nuance and I think we are going to be tweaking it for a while. It also seems nicely reusable.

Other C++ extensions that we might see:
- .cxx
- .C (really deep cut)

Reply:
Also, while we are thinking about it: would it make sense to warn on unrecognized file extensions if we are in this mode?

Reply:
Done, and generalized a bit more.